### PR TITLE
Auth Route Tests

### LINKED ISSUE #6: Implement Basic Authentication with Passport.js and JWT
**Description:**
As part of our application's authentication system, we need to implement basic authentication using Passport.js and JWT (JSON Web Tokens). This will allow us to securely authenticate users and manage their sessions using JWT tokens.

**Tasks:**
1. Set up Passport.js and JWT dependencies in the project.
2. Configure Passport.js with the JWT strategy to handle authentication.
3. Implement user registration endpoint to create new user accounts securely.
4. Create a login endpoint that authenticates users and generates JWT tokens.
5. Implement a protected endpoint that requires JWT token for access.
6. Set up middleware to authenticate requests using Passport.js and the JWT strategy.
7. Handle logout functionality to invalidate tokens or sessions.
8. Implement password reset functionality for users who have forgotten their password.

**Acceptance Criteria:**
- Users should be able to register and create new accounts securely.
- Users should be able to log in and receive a JWT token for subsequent authenticated requests.
- The protected endpoint should only be accessible with a valid JWT token.
- JWT tokens should be properly validated, and expired or invalid tokens should be rejected.
- Users should be able to log out and invalidate their session or token.
- The password reset functionality should allow users to reset their password securely.

**Additional Information:**
- Make sure to follow best practices for security, such as properly hashing passwords and securely handling JWT tokens.
- Document the authentication API endpoints and any required input parameters or response formats.
- Ensure that appropriate error handling is implemented for various scenarios, such as invalid credentials or expired tokens.

**Related Dependencies or References:**
- Passport.js: [link to documentation](https://www.passportjs.org/)
- JWT: [link to documentation](https://jwt.io/)

### PR DESCRIPTION
**Pull request by sathsarabandaraj**
## Purpose

The purpose of this PR is to fix #6 

## Goals

Test reliablitiy of auth endpoints

## Approach

Write test

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Related PRs
 
#13

### CODE REVIEW

File: src/services/auth.service.ts [hunk 1]
```diff
@@ -29,7 +29,19 @@ export const registerUser = async (
   })
   await profileRepository.save(newProfile)
 
-  return newProfile
```
Comment (anjula-sack) [inline src/services/auth.service.ts:32]: Why can we return newProfile? It's not returning the password right?

File: src/services/auth.service.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { JWT_SECRET } from '../configs/envConfig'
 export const registerUser = async (
   email: string,
   password: string
-): Promise<Profile> => {
+): Promise<Object> => {
```
Comment (anjula-sack) [inline src/services/auth.service.ts:10]:
```suggestion
): Promise<Profile> => {
```

File: src/services/auth.service.ts (hunk 1 above)
Comment (sathsarabandaraj) [reply src/services/auth.service.ts:32]: this is the result when using newProfile even when I set password select option to false

**Comment by anjula-sack** [comment]
@sathsarabandaraj can you fix the conflicts?

**Comment by anjula-sack** [comment]
any updates on this @sathsarabandaraj

**Comment by anjula-sack** [comment]
any updates on this @sathsarabandaraj

**Pull request closed without merging**